Tests/TuistKitTests/Generator/SchemesGeneratorTests.swift

```diff
@@ -14,13 +14,84 @@ final class SchemeGeneratorTests: XCTestCase {
+        let projectPath = AbsolutePath("/project.xcodeproj")
+        
+        let targets = ["App": pbxAppTarget,
+                       "AppTests": pbxAppTestsTarget]
+        let project = Project.test(targets: [app, appTests])
+        
+        let generatedProject = GeneratedProject(path: projectPath,
+                                                targets: targets)
+
+        let graphCache = GraphLoaderCache()
+        let graph = Graph.test(cache: graphCache)
+        
+        let got = subject.projectBuildAction(project: project,
+                                            generatedProject: generatedProject,
+                                            graph: graph)
+        
+        XCTAssertTrue(got.parallelizeBuild)
+        XCTAssertTrue(got.buildImplicitDependencies)
+        XCTAssertEqual(got.buildActionEntries.count, 2)
+        
+        let appEntry = got.buildActionEntries.first
+        let testsEntry = got.buildActionEntries.last
+        
+        XCTAssertEqual(appEntry?.buildFor, [.analyzing, .archiving, .profiling, .running, .testing])
+        XCTAssertEqual(appEntry?.buildFor, [.analyzing, .archiving, .profiling, .running, .testing])
+        XCTAssertEqual(appEntry?.buildableReference.referencedContainer, "container:project.xcodeproj")
+        XCTAssertEqual(appEntry?.buildableReference.buildableName, app.productName)
+        XCTAssertEqual(appEntry?.buildableReference.blueprintName, app.name)
+        XCTAssertEqual(appEntry?.buildableReference.buildableIdentifier, "primary")
+
+        XCTAssertEqual(testsEntry?.buildFor, [.testing])
+        XCTAssertEqual(testsEntry?.buildableReference.referencedContainer, "container:project.xcodeproj")
+        XCTAssertEqual(testsEntry?.buildableReference.buildableName, appTests.productName)
+        XCTAssertEqual(testsEntry?.buildableReference.blueprintName, appTests.name)
+        XCTAssertEqual(testsEntry?.buildableReference.buildableIdentifier, "primary")
+    }
+    
+    func test_projectTestAction() {
+        let app = Target.test(name: "App", product: .app)
+        let appTests = Target.test(name: "AppTests", product: .unitTests)
+        let pbxAppTarget = PBXNativeTarget(name: "App")
+        let pbxAppTestsTarget = PBXNativeTarget(name: "AppTests")
+        let projectPath = AbsolutePath("/project.xcodeproj")
+
+        let targets = ["App": pbxAppTarget,
+                       "AppTests": pbxAppTestsTarget]
+        let project = Project.test(targets: [app, appTests])
+        
+        let generatedProject = GeneratedProject(path: projectPath,
+                                                targets: targets)
+        let got = subject.projectTestAction(project: project,
+                                            generatedProject: generatedProject)
+        
+        XCTAssertEqual(got.buildConfiguration, "Debug")
+        XCTAssertNil(got.macroExpansion)
+        XCTAssertEqual(got.testables.count, 1)
```

Comment:
minor: 

It may be worth for test, including another test target e.g. a UITest target to test the logic of including all test targets.

Reply:
Good idea 👍